### PR TITLE
Implement rankings API models

### PR DESCRIPTION
**Pull request by EVAST9919**
Still without charts due to web reasons

### CODE REVIEW

**Comment by bdach** [comment]
What kind of web reasons are we talking about?

**Comment by EVAST9919** [comment]
> What kind of web reasons are we talking about?

https://github.com/ppy/osu-web/issues/4898

File: osu.Game.Tests/Visual/Online/TestSceneRankingsApi.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+using osu.Game.Rulesets.Taiko;
+using osu.Game.Rulesets.Catch;
+using osu.Game.Graphics.UserInterface;
+
+namespace osu.Game.Tests.Visual.Online
+{
+    public class TestSceneRankingsApi : OsuTestScene
+    {
+        [Resolved]
+        private IAPIProvider api { get; set; }
```
Comment (peppy) [inline osu.Game.Tests/Visual/Online/TestSceneRankingsApi.cs]: Should be using `API` instead, with `UseOnlineAPI => true`

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Online/API/Requests/GetRankingsRequest.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            this.ruleset = ruleset;
+            this.page = page;
+        }
+
+        protected override WebRequest CreateWebRequest()
+        {
+            var req = base.CreateWebRequest();
+
+            req.AddParameter("page", page.ToString());
```
Comment (smoogipoo) [inline osu.Game/Online/API/Requests/GetRankingsRequest.cs]: I think this is fine for now as this isn't the final API.

**Comment by EVAST9919** [comment]
Closing because it's already included in https://github.com/ppy/osu/pull/6293 and there's no a good way to test it without ui (which is already included in https://github.com/ppy/osu/pull/6293)

**Pull request closed without merging**